lib/cadet/updater/CS1101S.ex

```diff
@@ -0,0 +1,69 @@
+defmodule Cadet.Updater.CS1101S do
+  @moduledoc """
+  Pulls content from the remote cs1101s repository into a local one.
+  """
+
+  @remote_repo (if Mix.env() != :test do
+                  Dotenv.load().values["01S_REPOSITORY"]
+                else
+                  "test/remote_repo"
+                end)
+  @local_name if Mix.env() != :test, do: "cs1101s", else: "test/local_repo"
+  @key_file Dotenv.load().values["01S_RSA_KEY"]
+
+  require Logger
+
+  def clone() do
+    Logger.info("Cloning CS1101S: Started")
+    git("clone", [@remote_repo, @local_name])
+    Logger.info("Cloning CS1101S: Done")
+  end
+
+  def update() do
+    Logger.info("Updating CS1101S...")
+    git("fetch")
+    git("pull")
+  end
+
+  def fetch() do
+    git("fetch")
+  end
+
+  def pull() do
+    git("pull")
+  end
+
+  defp git(cmd, args \\ [])
+
+  defp git("clone", args) do
+    {out, exit} =
+      System.cmd(
+        "git",
+        ["clone"] ++ args,
```

Comment:
this one does not have the `-C @local_name` argument?

Reply:
Nope, it does something like `$ git clone git@github.com/cs1101s/cs1101s.git .` which creates a new directory `cs1101s` in cwd, which is the mix project root in dev/test and wherever you run the cadet binary from in prod.

Reply:
Oh rightttt. Sorry I derped